service/controller/v2/resource/resourcegroup/resource.go

```diff
@@ -105,21 +107,30 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring resource group is deleted")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring resource group deletion")
 
-	f, err := groupsClient.Delete(ctx, key.ClusterID(customObject))
+	_, err = groupsClient.Get(ctx, key.ClusterID(customObject))
 	if IsNotFound(err) {
 		// fall through
 	} else if err != nil {
 		return microerror.Mask(err)
 	} else {
-		err = f.WaitForCompletion(ctx, groupsClient.Client)
-		if err != nil {
+		_, err := groupsClient.Delete(ctx, key.ClusterID(customObject))
```

Comment:
What is the effect of triggering `Delete` for a resource group in deleting state ?

Reply:
This was discussed already in the other comments. The action seems to be idempotent which was what I expected in the first place.

Reply:
Cool.

Sorry for duplicate, previous comments did not showed up in diff.